ddprof-lib/src/main/cpp/flightRecorder.cpp

```diff
@@ -367,6 +368,8 @@ off_t Recording::finishChunk(bool end_recording) {
 }
 
 void Recording::switchChunk(int fd) {
+    // ClearTask is a RAAI that will switch the epoch immediately and release the cached references when going out of scope
+    ClearTask cc = ClassRefCache::instance()->clear();
```

Comment:
See above, I think the `ClearTask` needs to be constructed from an acquired epoch here to achieve the desired semantics.